### PR TITLE
Add free space checks on slab expansion (SYN-4615)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -1012,6 +1012,7 @@ async def __anit__(self, dirn, conf=None, readonly=False, parent=None):
         self.isactive = False
         self.inaugural = False
         self.activecoros = {}
+        self._checkspace = None
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:1015]: Is there a reason to not initialize this as an asyncio.Eeent directly?

File: synapse/lib/lmdbslab.py [hunk 2]
```diff
@@ -941,6 +945,7 @@ def _growMapSize(self, size=None):
         self.mapsize = mapsize
 
         self.resizeevent.set()
+        [callback() for callback in self.resizecallbacks]
```
Comment (vEpiphyte) [inline synapse/lib/lmdbslab.py:948]: Do we want to do anything about these callbacks possible throwing an exception? Feels like a bad place to break down from a random callback.

Comment (vEpiphyte) [reply synapse/lib/lmdbslab.py:948]: Can we make the above logger.warning() call a logger.info call while we are changing this code? That warning level is an artifact of the original lmdb implementation and is not an indicator of something amiss.

File: synapse/lib/cell.py (hunk 1 above)
Comment (Cisphyx) [reply synapse/lib/cell.py:1015]: Not really, we might not need it but one extra event probably isn't a big deal

File: synapse/lib/lmdbslab.py (hunk 2 above)
Comment (Cisphyx) [reply synapse/lib/lmdbslab.py:948]: Yeah good point, will catch and log exceptions here

File: synapse/lib/cell.py (hunk 1 above)
Comment (vEpiphyte) [reply synapse/lib/cell.py:1015]: asyncio events are very cheap ( unlike threading events ) so we can make them without issue.

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**